code/__DEFINES/is_helpers.dm

```diff
@@ -209,6 +209,8 @@ GLOBAL_LIST_INIT(heavyfootmob, typecacheof(list(
 
 #define ismachinery(A) (istype(A, /obj/machinery))
 
+#define ishydroponics(A) (istype(A, /obj/machinery/hydroponics))
```

Comment:
is this the best way of doing what I am trying to accomplish?

Reply:
Helpers are for commonly used is types, I'm not sure if hydroponics trays checks are common enough to warrant this. Either way you should replace all typechecks for hydroponics trays with this helper if you really want to use a helper.

Reply:
Thanks ima remove it, I didnt know if it was cheaper to do or something for language reasons

Reply:
for future reference, defines are interpreted by the compiler as what its defined at, so to the compiler  ishydroponics(A) just means istype(A, /obj/machinery/hydroponics) so theres no runtime difference.